src/aat/resources/features/F-122 - Conditional Event Post States/F-122.feature

```diff
@@ -0,0 +1,150 @@
+
+    Then a positive response is received,
+    And the response [contains state: CaseRevoked, updated values for AddressField.AddressLine1, EmailField along with an HTTP-201 Created],
+
+#-----------------------------------------------------------------------------------------------------------------------
+  @S-122.5
+  Scenario: Ordering should take precedence when both conditions resolve (End state: CaseUpdated2)
+
+    Given a user with [an active profile in CCD],
+    And a successful call [to create a token for case creation] as in [S-122-GetToken_CaseCreate],
+    And a successful call [to create a case] as in [FT_ConditionalPostState_Create_Case],
+    And a successful call [to get an event token for the case just created] as in [S-122-GetToken_UpdateCase],
+
+    When a request is prepared with appropriate values,
+    And the request [contains a case Id that has just been created],
+    And the request [contains Update token created as in S-122-GetToken_UpdateCase],
+    And it is submitted to call the [submit updateCase2 event with TextField and EmailField values] operation of [CCD data store],
+
+    Then a positive response is received,
+    And the response [contains state: CaseUpdated2, updated values for TextField, EmailField along with an HTTP-201 Created],
+
+#-----------------------------------------------------------------------------------------------------------------------
+  @S-122.6
+  Scenario: Sets the state defined with a matching post state condition using OR operator (End state: CaseAmended2)
+
+    Given a user with [an active profile in CCD],
+    And a successful call [to create a token for case creation] as in [S-122-GetToken_CaseCreate],
+    And a successful call [to create a case] as in [FT_ConditionalPostState_Create_Case],
+    And a successful call [to get an event token for the case just created] as in [S-122-GetToken_UpdateCase],
+
+    When a request is prepared with appropriate values,
+    And the request [contains a case Id that has just been created],
+    And the request [contains Update token created as in S-122-GetToken_UpdateCase],
+    And it is submitted to call the [submit updateCase2 event with TextField and EmailField values] operation of [CCD data store],
+
+    Then a positive response is received,
+    And the response [contains state: CaseAmended2, updated values for TextField, EmailField along with an HTTP-201 Created],
+
+#-----------------------------------------------------------------------------------------------------------------------
+  @S-122.7
+  Scenario: *(FieldA) will keep the state as is (End state: CaseCretated)
+
+    Given a user with [an active profile in CCD],
+    And a successful call [to create a token for case creation] as in [S-122-GetToken_CaseCreate],
+    And a successful call [to create a case] as in [FT_ConditionalPostState_Create_Case],
+    And a successful call [to get an event token for the case just created] as in [S-122-GetToken_UpdateCase],
```

Comment:
"${[scenarioContext][parentContext][childContexts][S-122-GetToken_CaseCreate][testData][actualResponse][body][token]}"

Reply:
Sorry, Please ignore this, as this is a new update event, we need to generate the event_token again

Reply:
Sorry, Please ignore this, as this is a new update event, we need to generate the event_token again